### ISSUE TITLE
Problem in MatchMaker.remoteRoomCall

### ISSUE THREAD

**Issue by holywyvern** [gamestdio/colyseus#140]
I was investigating a bit using clusters on my Colyseus server.

I found when I was using a cluster, just after `requestJoin `and before `onAuth` The client never joined.
On the log, I found than the error "remote room timed out" was thrown.
After a bit of finding and testing, I found the problem:

When the room is called in MatchMaker.remoteRoomCall the room id is: "147,10,170,66,107,101,79,81,116,121,104,105,102,1" or something like that, but when it publishes the message it is a room id like "HkuXKy2jM" instead, making getRoomChannel to return wildly different values.

I'll try to isolate my bit of code than is causing troubles and post an update. I tried using both RedisPresence and MemsharedPresence, so it seems it's not a problem on either of them.

**Comment by endel** [comment]
Hi @holywyvern, can you confirm which version of Node.js and operational system you're using?

**Comment by holywyvern** [comment]
Windows 10 x86-65
Node.js: 8.9.4
I'm on my development machine, didn't test on production.

**Comment by endel** [comment]
@holywyvern I've noticed an odd behavior when locking the rooms. I've just released version [`0.9.5`](https://github.com/gamestdio/colyseus/releases/tag/0.9.5) which __may__ fix this issue. Let me know if that works for you!

**Comment by seiyria** [comment]
Also having this issue, but the repro case is slightly more complex in my case. Thanks for boiling this down!

**Comment by bamsutler** [comment]
I am seeing a similar error without the clustering setup. 



```
Error: remote room timed out
    at Timeout.setTimeout (<my path>/node_modules/colyseus/lib/MatchMaker.js:149:32)
    at ontimeout (timers.js:482:11)
    at tryOnTimeout (timers.js:317:5)
    at Timer.listOnTimeout (timers.js:277:5)
```

I am running this on my local machine with a defold client. I have this for the server. 

```
import * as path from 'path';
import * as express from 'express';
import * as serveIndex from 'serve-index';
import { createServer } from 'http';
import { Server } from 'colyseus';
import { monitor } from '@colyseus/monitor';
import * as WebSocket from 'uws';

// Require ChatRoo handler
 import { ChatRoom } from "./rooms/ChatRoom";
// import { StateHandlerRoom } from "./rooms/02-state-handler";
// import { AuthRoom } from "./rooms/03-auth";
// import { CreateOrJoinRoom } from "./rooms/04-create-or-join-room";

const port = Number(process.env.PORT || 2727);
const app = express();

// Attach WebSocket Server on HTTP Server.
const gameServer = new Server({
  engine: WebSocket.Server,
  server: createServer(app)
});

// Register ChatRoom as "chat"
gameServer.register("chat", ChatRoom)
  .on("create", (room) => console.log("room created:", room.roomId))
  .on("dispose", (room) => console.log("room disposed:", room.roomId))
  .on("join", (room, client) => console.log(client.id, "joined", room.roomId))
  .on("leave", (room, client) => console.log(client.id, "left", room.roomId));

// Register ChatRoom with initial options, as "chat_with_options"
// onInit(options) will receive client join options + options registered here.
gameServer.register("chat_with_options", ChatRoom, {
    custom_options: "you can use me on Room#onInit"
});

// Register StateHandlerRoom as "state_handler"
// gameServer.register("state_handler", StateHandlerRoom);

// Register StateHandlerRoom as "state_handler"
// gameServer.register("auth", AuthRoom);

// Register CreateOrJoin as "create_or_join"
// gameServer.register("create_or_join", CreateOrJoinRoom);

app.use('/', express.static(path.join(__dirname, "static")));
app.use('/', serveIndex(path.join(__dirname, "static"), {'icons': true}))

// (optional) attach web monitoring panel
app.use('/colyseus', monitor(gameServer));

gameServer.listen(port);
gameServer.onShutdown(function(){
  console.log(`game server is going down.`)
})
gameServer.matchMaker.create("chat", {})

gameServer.matchMaker.getAllRooms('chat').then(
  function(rooms){ console.log(rooms)}
)
console.log(`Listening on http://localhost:${ port }`);
```

which is effectively the demo on the colyseus.io website, except I am forcing the creation of the room which is set to not auto-dispose. 

The client is similarly taken from the examples. 
```
local client 
local room 

function init(self)

	client = ColyseusClient.new("ws://localhost:2727");

	client:on('open', function()
		pprint("connection is now open")
	end)

	client:on('close', function()
		print("connection has been closed")
		
	end)

	client:on("error", function()
		print("something wrong happened")
	end)

	-- join chat room
	room = client:join("chat")
	
end

function update(self, dt)
	-- Add update code here
	-- Remove this function if not needed
	client:loop()
end

```

Seeing the diffrence in the room ids in this ticket i logged the room id in the failing method and the room id that i was creating and got this. 

```
BasicRoom created! {}
room created: Bk0Kz-kyQ
Listening on http://localhost:2727
[ { clients: 0,
    maxClients: 4,
    metadata: undefined,
    roomId: 'Bk0Kz-kyQ' } ]
B12JCn6Cf in the bowls
Error: remote room timed out
    at Timeout.setTimeout (<my path>/node_modules/colyseus/lib/MatchMaker.js:150:32)
    at ontimeout (timers.js:482:11)
    at tryOnTimeout (timers.js:317:5)
    at Timer.listOnTimeout (timers.js:277:5)
```

as you can see, the roomId in the MatchMaker.js file is B12JCn6Cf where the roomId I created is Bk0Kz-kyQ. Oddly that was the first run and I didn't want to post "In the bowls" and tried to change that comment to the file name and got the same roomId in a completely separate build. 

```
[nodemon] starting `ts-node src/index.ts`
BasicRoom created! {}
room created: ryJOVbJJX
Listening on http://localhost:2727
[ { clients: 0,
    maxClients: 4,
    metadata: undefined,
    roomId: 'ryJOVbJJX' } ]
B12JCn6Cf in MatchMaker.js
Error: remote room timed out
    at Timeout.setTimeout (<my path>/node_modules/colyseus/lib/MatchMaker.js:150:32)
    at ontimeout (timers.js:482:11)
    at tryOnTimeout (timers.js:317:5)
    at Timer.listOnTimeout (timers.js:277:5)
```

I have added some logging to try to find where that ID id comming from with no luck so far.
```
[nodemon] starting `ts-node src/index.ts`
HJclOb11Q in create
BasicRoom created! {}
HJclOb11Q in createRoomReferences
room created: HJclOb11Q
Listening on http://localhost:2727
HJclOb11Q in getAvilableRooms
HJclOb11Q in remoteRoomCallTimer
[ { clients: 0,
    maxClients: 4,
    metadata: undefined,
    roomId: 'HJclOb11Q' } ]
B12JCn6Cf in remoteRoomCallTimer
B12JCn6Cf in remoteRoomCallTimer
Error: remote room timed out
    at Timeout.setTimeout (<mypath>/node_modules/colyseus/lib/MatchMaker.js:153:32)
    at ontimeout (timers.js:482:11)
    at tryOnTimeout (timers.js:317:5)
    at Timer.listOnTimeout (timers.js:277:5)
^CDispose BasicRoom
room disposed: HJclOb11Q
game server is going down.
```

Going to keep looking at this. Let me know if you have any thoughts from what I posted so far. 

node v8.11.1
Mac Osx 10.13.4 (17E202)
colyseus 0.9.5

**Comment by bamsutler** [comment]
With a little more debug logging, the remote call that is triggering the error is `getInspectData`. I could not find this remote call anywhere in the colyseus project and decided to disable the only other thing calling the room's information ( I did not have the client running ) the monitor. This seems to fix the error. 


Turns out mine was mostly unrelated to joining rooms.

**Comment by endel** [comment]
Hey @bamsutler, that's really odd. I've just tried the [code you pasted](https://github.com/gamestdio/colyseus/issues/140#issuecomment-390487661) and my logs look like this:

```
BasicRoom created! {}
room created: BkaM1VJ1X
Listening on http://localhost:2727
[ { clients: 0,
    maxClients: 4,
    metadata: undefined,
    roomId: 'BkaM1VJ1X' } ]
BJIXTmy17 joined BkaM1VJ1X
```

The `getInspectData` method comes from `colyseus-monitor` project. My guess is that you had the monitor open viewing a room on a tab, and then rebooted the server - then the monitor would try to retrieve data for an unexisting room, could that be the case?

Cheers!

**Comment by endel** [comment]
I'm closing this since it seems to be fixed on `0.9.16`. Let me know if you still see this error!
